src/test/java/edu/ucsb/mapache/controllers/SearchHistoryControllerTests.java

```diff
@@ -0,0 +1,145 @@
+    private String authorization = "Bearer: " + jwtValue;
+  private ObjectMapper mapper = new ObjectMapper();
+  @Autowired
+  private MockMvc mockMvc;
+   
+  @Autowired
+  private SearchHistoryController searchhistorycontoller;
+  
+  @MockBean
+  AppUserRepository appUserRepository;
+  @MockBean
+  GoogleSearchService googleSearchService;
+  @MockBean
+  AuthControllerAdvice authControllerAdvice;
+
+  @MockBean
+  private SearchRepository searchRepository;
+
+  @MockBean
+  private UserSearchRepository userSearchRepository;
+
+  @MockBean
+  private PropertiesService propertiesService;
+
+  private AppUser getMockAppUser() {
+    AppUser appUser = new AppUser();
+    appUser.setId(1);
+    appUser.setEmail("haixinlin123@umail.ucsb.edu");
+    appUser.setFirstName("Hunter");
+    appUser.setLastName("Lin");
+    appUser.setSearchRemain(100);
+    appUser.setTime(0);
+    appUser.setApiToken("fake-but-valid-api-token");
+    return appUser;
+  }
+
+  @Test
+  public void test_getJWT() {
+        DecodedJWT actual = searchhistorycontoller.getJWT(authorization);
+        // Writing the code twice is not the best way to do a test.
+        // In this case, I don't yet have a better alterantive to propose.
+        DecodedJWT expected = JWT.decode(authorization.substring(7));
+        // Note that the .equals for DecodedJWT is broken, so we need this instead:
+        assertEquals(expected.getToken(), actual.getToken());
+    }
+  
+  @Test
+  public void test_basicSearch_unauthorizedIfNotMember() throws Exception {
+    when(propertiesService.getNamespace()).thenReturn("https://proj-mapache-search.herokuapp.com");
+    mockMvc.perform(get("/api/members/searchhistory/allusersearches").contentType("application/json")
+        .header(HttpHeaders.AUTHORIZATION, exampleAuthToken)).andExpect(status().is(401));
+  }
+
+  @Test
+  public void test_quota_unauthorizedIfNotMember() throws Exception {
+
+    // TODO CHANGE TO APPROPRIATE URL FOR THIS CONTROLLER
+
+    // mockMvc
+    // .perform(
+    // get("/api/member/search/quota").contentType("application/json").header(HttpHeaders.AUTHORIZATION,
+    // exampleAuthToken))
+    // .andExpect(status().is(401));
+  }
+
+  @Test
+  public void test_getSearches_case1() throws Exception {
+
+    // write code to test the method
+    //
+    // @GetMapping("/allusersearches")
+    // public ResponseEntity<String> getSearches(@RequestHeader("Authorization")
+    // String authorization)
+
+    // I wrote 'case 1' because you'll probably need a few cases, e.g.
+    // for not logged in, logged in as admin, logged in as member, etc.
+    // depending on what the code is checking for
+  }
+
+  @Test
+  public void test_getSearches_case2() throws Exception {
```

Comment:
I am really confused why there are three different PRs that seem to be covering the same territory, and why there is commented out code still in two of them. 

Should these be closed or marked as drafts?